From 9bcba132fa59332aa274620e081baceaa7731945 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 16 Dec 2015 03:51:20 +0000
Subject: [PATCH] stdout/stderr are const objects cant assign to them like this

flush stderr and stdout then assign /dev/null to them

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/shared/pager.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/src/shared/pager.c b/src/shared/pager.c
index 3bec27b..9377501 100644
--- a/src/shared/pager.c
+++ b/src/shared/pager.c
@@ -155,13 +155,20 @@ int pager_open(bool no_pager, bool jump_to_end) {
 
 void pager_close(void) {
 
+        int fd;
         if (pager_pid <= 0)
                 return;
 
         /* Inform pager that we are done */
-        stdout = safe_fclose(stdout);
-        stderr = safe_fclose(stderr);
-
+        fflush(stdout);
+        fd = open("/dev/null", O_RDWR);
+	if (fd != -1)
+                dup2(fd, 1);
+        fflush(stderr);
+        fd = open("/dev/null", O_RDWR);
+	if (fd != -1)
+                dup2(fd, 2);
+        close(fd);
         (void) kill(pager_pid, SIGCONT);
         (void) wait_for_terminate(pager_pid, NULL);
         pager_pid = 0;
